deserialize/__init__.py

```diff
@@ -37,15 +37,19 @@ def _deserialize(class_reference, data, debug_name):
         for valid_type in valid_types:
             try:
                 return _deserialize(valid_type, data, debug_name)
-            except:
+            except DeserializeException:
                 pass
+        raise DeserializeException(f"Cannot deserialize '{type(data)}' to '{class_reference}' for '{debug_name}'")
 
     if isinstance(data, dict):
         return _deserialize_dict(class_reference, data, debug_name)
 
     if isinstance(data, list):
         return _deserialize_list(class_reference, data, debug_name)
 
+    if isinstance(data, tuple) and is_typing_type(class_reference):
```

Comment:
You shouldn't need the ` and is_typing_type(class_reference)` since it's handled in the `_deserialize_tuple` method anyway.

Reply:
This is supposed to differentiate between plain python `tuple` and `typing.Tuple`. Won't work without.

Reply:
Ah, good point. We should probably handle the decode in that case as well, similar to how you've added support for plain `dict`?